Orange/misc/cache.py

```diff
@@ -41,20 +41,16 @@ def memoize_method(*lru_args, **lru_kwargs):
     """
     def _decorator(func):
 
+        @lru_cache(*lru_args, **lru_kwargs)
+        def _cached_method(self_weak, *args, **kwargs):
+            return func(self_weak(), *args, **kwargs)
+
         @wraps(func)
         def _wrapped_func(self, *args, **kwargs):
-            self_weak = weakref.ref(self)
-            # We're storing the wrapped method inside the instance. If we had
-            # a strong reference to self the instance would never die.
```

Comment:
Please explain in the docstring why is this needed and how is it different from lru_cache.